samples: matter: Refactored samples configuration structure (#7497)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

  * :ref:`matter_lock_sample` sample:

    * Added support for the Door Lock cluster, which replaced the previous temporary solution based on the On/Off cluster.
  

>   * Simplified all samples :file:`prj.conf` files by using default configuration from Matter upstream :file:`Kconfig.defaults` file.

Reviewer: Move this to line 274.
```suggestion
  * Simplified the :file:`prj.conf` file of each sample by using the default configuration from the :file:`Kconfig.defaults` file in Matter upstream.
```

The following list summarizes the most important changes inherited from the upstream Matter:

* Added the Binding cluster and Groupcast communication to the Light Switch sample.

> * Added for nRF Connect SDK platform :file:`Kconfig.defaults`, :file:`Kconfig.mcuboot.defaults` and :file:`Kconfig.multiprotocol_rpmsg.defaults` files containing default configuration common for all nRF Connect SDK samples.

Reviewer: ```suggestion
* Updated the nRF Connect platform by adding :file:`Kconfig.defaults`, :file:`Kconfig.mcuboot.defaults` and :file:`Kconfig.multiprotocol_rpmsg.defaults` files that contain the default configuration for all nRF Connect SDK samples.
```
The platform in Matter is called `nRF Connect` last time I checked, not `nRF Connect SDK`, but maybe you meant something else here than my suggestion? Let me know.
